### ISSUE TITLE
Regression: empty _delete.pid directories left in place when use_lazy_deletes

### ISSUE THREAD

**Issue by sgpinkus** [rsnapshot/rsnapshot#282]
Introduced in b8f33b922dfd3b79654312882a5ba8c9cf572256. Ref #262.

How to acutally reproduce the error in #262: 

  - Only occurs with `sync_first` off.
  - Set rsnapshot running and `killall rsync` so it exist with a certain error code (should exit with 20).

Test config:

```
config_version	1.2
snapshot_root	/tmp/rsnapshot-test/backups
lockfile	/tmp/rsnapshot-test/rsnapshot-test.lock
logfile	/tmp/rsnapshot-test/rsnapshot.log
cmd_cp		/bin/cp
cmd_rm		/bin/rm
cmd_rsync	/usr/bin/rsync
cmd_logger	/usr/bin/logger
cmd_ssh	/usr/bin/ssh
retain		alpha	2
retain		beta	2
sync_first	0
link_dest	1
verbose	3
loglevel	3
backup	/tmp/rsnapshot-test/src		./
```

Test invokation:

```
cd /tmp//tmp/rsnapshot-test/
(mkdir src/ && cd src/ &&  make_test_file_tree.sh 2 25 25000) # Make largish amount of data to backup so can kill rsnapshot while running
~/src/rsnapshot/rsnapshot -c 262.conf alpha & echo $! && (sleep 0.5; killall rsync)
~/src/rsnapshot/rsnapshot -c 262.conf alpha & echo $! && (sleep 0.5; killall rsync) # Do more than once from fresh.
```

**Comment by sgpinkus** [comment]
Actually noticed it's not just _delete.pid dirs, empty backup dirs are left around too,

**Comment by sgpinkus** [comment]
Fixed in #283.
